projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -324,69 +360,35 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
   public onFocus(): void {
     this._isFocused = true;
     this.onTouchCallback();
-    this.updateMask();
+    if (this.value) {
+      this.updateMask();
+    } else {
+      this.inputValue = this.emptyMask;
+    }
     super.onFocus();
   }
 
   /** @hidden @internal */
   public onBlur(value: string): void {
     this._isFocused = false;
-
-    if (this.inputValue === this.emptyMask) {
-      if (this.value) {
-        this.updateValue(null);
-      }
+    if (!this.inputIsComplete() && this.inputValue !== this.emptyMask) {
+      this.updateValue(this.parseDate(this.inputValue));
+    }
+    if (!this.isValidDate(this.value)) {
       this.inputValue = '';
       return;
     }
 
-    const parsedDate = this.parseDate(this.inputValue);
-    if (this.isValidDate(parsedDate)) {
-      this.updateValue(parsedDate);
-    } else {
-      const oldValue = this.value && new Date(this.value.getTime());
-      const args = { oldValue: oldValue, newValue: parsedDate, userInput: this.inputValue };
-      this.validationFailed.emit(args);
-      if (args.newValue?.getTime && args.newValue.getTime() !== oldValue.getTime()) {
-        this.updateValue(args.newValue);
-      } else {
-        this.updateValue(null);
-      }
-    }
-
     this.updateMask();
     super.onBlur(value);
   }
 
   /** @hidden @internal */
   public updateMask(): void {
-    if (!this.value || !this.isValidDate(this.value)) {
-      this.inputValue = this._isFocused ? this.emptyMask : '';
-      return;
-    }
     if (this._isFocused) {
+      // store the cursor position as it will be moved during masking
       const cursor = this.selectionEnd;
-      let mask = this.emptyMask;
-      for (const part of this._inputDateParts) {
-        if (part.type === DatePart.Literal) { continue; }
-
-        const partLength = part.format.length;
-        let targetValue = this.getPartValue(part.type, partLength);
-        if (part.type === DatePart.Month) {
-          targetValue = this.prependValue(
-            parseInt(targetValue.replace(new RegExp(this.promptChar, 'g'), '0'), 10) + 1, partLength, '0');
-        }
-        if (part.type === DatePart.Hours && part.format.indexOf('h') !== -1) {
-          targetValue = this.prependValue(this.toTwelveHourFormat(targetValue), partLength, '0');
-        }
-        if (part.type === DatePart.Year && partLength === 2) {
-          targetValue = this.prependValue(parseInt(targetValue.slice(-2), 10), partLength, '0');
-        }
-
-        mask = this.maskParser.replaceInMask(mask, targetValue, this.maskOptions, part.start, part.end).value;
-      }
-
-      this.inputValue = mask;
+      this.inputValue = this.getMaskedValue();
       this.setSelectionRange(cursor);
     } else {
       const format = this.displayFormat || this.inputFormat;
```

Comment:
if it is not focused check if we have value - if not set `inputValue` to empty string and return